### ISSUE TITLE
Cannot load component

### ISSUE THREAD

**Issue by hemmingsentfromhell** [kpoppel/homeassistant-eforsyning#20]
Version:
Home Assistant Core 2022.6.7
Home Assistant Supervisor 2022.05.3
Home Assistant OS 8.2

Added custom repository:
https://github.com/kpoppel/homeassistant-eforsyning

Installed and rebooted.

Enterred username (e-mail) and password that I login with on eforsyning and I found my supplier ID: (some are masked with X)
E80XXXXX-5336-XXXX-A1XX-461XXXX81349

Getting this error when adding the integration to Home Assistant:

2022-07-04 21:37:22 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry Eforsyning E80XXXXX-5336-XXXX-A1XX-461XXXX81349 for eforsyning
File "/config/custom_components/eforsyning/__init__.py", line 43, in async_setup_entry

**Comment by kpoppel** [comment]
Try and check and uncheck the billing period checkbox. Then it should work. I don't know how to get that part right.
If this works, please close the issue again.

**Comment by hemmingsentfromhell** [comment]
Thank you for the tip. That worked and no errors come up. 
But all sensors are listed as 'unknown'. Have you seen this before?

Logger: custom_components.eforsyning
Source: custom_components/eforsyning/__init__.py:109
Integration: EForsyning
First occurred: 11:00:18 (1 occurrences)
Last logged: 11:00:18

Exception: <class 'KeyError'>

**Comment by kpoppel** [comment]
KeyError is from some unexpected data returned from the server. Please take a look in the readme file to enable debugging information. This will dump out some more information on what it could be.
I really need to build in some robustness in the error handling 🤔

**Comment by hemmingsentfromhell** [comment]
I tried enabling the debugging, but no further information is displayed. :-(

**Comment by kpoppel** [comment]
Hmm. You should get a lot more information, like "logging in" and so on in the log file. If you cannot get the debugging to dump out more and don't mind sharing your login with me, I can take a look.  It will be in a couple of weeks in that case - Summer and so on 😊

**Comment by Dikymon** [comment]
I'm facing the same issue; This is all the logging i get with logging enabled.

`This error originated from a custom integration.

Logger: custom_components.eforsyning
Source: custom_components/eforsyning/__init__.py:109
Integration: EForsyning
First occurred: 9:54:14 PM (1 occurrences)
Last logged: 9:54:14 PM

Exception: <class 'TypeError'>`

**Comment by kpoppel** [comment]
Okay so something is different for you. I am on Core 2022.6.7 too, and see no issues with my data feed. To investigate further I would need some more data from the communication on your end(s).

**Comment by Dikymon** [comment]
I am on:
Home Assistant 2022.7.3
Supervisor 2022.07.0
Operating System 8.2
Frontend 20220707.0 - latest

Unsure how to log communication from the component?

Findings during install:
With june/july enabled makes it work, it seems.
Toggling june/july installs but gives all 'unknown' sensor values
Not touching the checkbox get you the async_setup_entry error.

**Comment by hemmingsentfromhell** [comment]
Hi again. 

I am using Home Assistant Core 2022.6.7

@Dikymon Mine doesn't work with june/july enabled. Still get unknown sensors.  

@kpoppel : I can send you my login for debugging purposes. Where can I send it to?

**Comment by kpoppel** [comment]
I have tested the integration on the latest Core and Supervisor now.
In a few minutes I will release v0.9.0 which includes improved config flow and the ability to have more than one supplier.
I have also verified that putting in these lines in `configuration.yaml` really does dump out a lot more information if one makes it past the configuration step (which everyone should be able to now):

```
logger:
  default: info
  logs:
    custom_components.eforsyning: debug
```

@hemmingsentfromhell I have sent you an invite to discuss some more on your issue.
